src/token_cache.rs

```diff
@@ -66,12 +68,26 @@ impl TokenCache {
                         jwt::header::Header,
                         jwt::claims::Claims,
                         jwt::token::Unverified,
-                    >::parse_unverified(token_str)
+                    >::parse_unverified(&token_str)
                     {
                         Ok(token) => token.claims().registered.expiration.unwrap_or(u64::MAX),
                         Err(error) => {
-                            warn!(?error, "Invalid bearer token");
-                            return;
+                            // the token doesn't have a claim that states a
```

Comment:
This logic looks good but assumes that the error returned is due to the missing expiry. This `Err` block should contain a guard using `matches!` or whatever else is needed to make sure we are handling only the error case of a missing expiry. Other error cases should continue with the warning like what existed before

Reply:
Good point, I've fixed that now